docs/tutorial_build_automation.rst

```diff
@@ -0,0 +1,183 @@
+For this reason it sets the relevant EasyBuild environment variables.
+
+.. tip::
+
+   Users may set the EasyBuild prefix to a different location by setting the :attr:`~reframe.core.buildsystems.EasyBuild.prefix` attribute of the build system.
+   This allows you to have the built software installed upon successful completion of the build phase, but if the test fails in a later stage (sanity, performance), the installed software will not be cleaned up automatically.
+
+.. note::
+
+   ReFrame assumes that the ``eb`` executable is available on the system where the compilation is run (typically the local host where ReFrame is executed).
+
+
+Now that we know everything related to building and installing the code, we can move to the part dealing with running it.
+To run the code, the generated modules need to be loaded in order to make the software available.
+The modules can be accessed through :attr:`~reframe.core.buildsystems.Easybuild.generated_modules`, however, they are available only after EasyBuild completes the installation.
+This means that :attr:`~reframe.core.pipeline.RegressionTest.modules` can be set only after the build phase finishes.
+For that, we can set :attr:`~reframe.core.pipeline.RegressionTest.modules` in a class method wrapped by the :py:func:`~reframe.core.RegressionTest.run_before` built-in, specifying the ``run`` phase.
+This test will then run the following commands:
+
+.. code-block:: console
+
+   module load bzip/1.0.6
+   bzip2 --help
+
+
+--------------------------
+Packaging the installation
+--------------------------
+
+The EasyBuild build system offers a way of packaging the installation via EasyBuild's packaging support.
+To use this feature, `the FPM package manager <https://fpm.readthedocs.io/en/latest/intro.html>`__ must be available.
+By setting the dictionary :attr:`~reframe.core.buildsystems.Easybuild.package_opts` in the test, ReFrame will pass ``--package-{key}={val}`` to the EasyBuild invocation.
+For instance, the following can be set to package the installations as an rpm file:
+
+.. code-block:: python
+
+   self.keep_files = ['easybuild/packages']
+   self.build_system.package_opts = {
+       'type': 'rpm',
+   }
+
+The packages are generated by EasyBuild in the stage directory.
+To retain them after the test succeeds, :attr:`~reframe.core.pipeline.RegressionTest.keep_files` needs to be set.
+
+
+
+Using Spack to Build the Test Code
+----------------------------------
+
+
+.. versionadded:: 3.6.1
+
+
+This example is the equivalent to the previous one, except that it uses Spack to build ``bzip2``.
+Here is the test's code:
+
+.. literalinclude:: ../tutorials/build_systems/spack/spack_test.py
+   :lines: 6-
+   :emphasize-lines: 12,14-16
+
+
+When :attr:`~reframe.core.pipeline.RegressionTest.build_system` is set to ``'Spack'``, ReFrame will leverage Spack environments in order to build the test code.
+For this reason, currently user must specify an environment.
```

Comment:
```suggestion
For this reason, the current user must specify an environment.
```

Reply:
Here, I do mean "currently", in terms of time. Because there is also this PR: #2031.

Reply:
I would write:
```
For this reason, users must specify an environment.
```

Reply:
You are right it confuses as it is written now. In fact, I wanted to write this:

```suggestion
For this reason, currently, users must specify an environment.
```